components/NoteEditor/Editor/Editor.js

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import dynamic from 'next/dynamic';
+
+import shortid from 'shortid';
+import styles from './Editor.module.scss';
+import Heading from './Heading';
+
+const Section = dynamic(() => import('./Section'), { ssr: false });
+
+function formatDate(dateObj) {
+  return new Intl.DateTimeFormat('en-US', { month: 'short', day: 'numeric' }).format(dateObj);
+}
+
+const formatedDate = formatDate(new Date()).split(' ');
+
+export default function Editor() {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [sections, setSections] = useState([{ id: shortid.generate(), keyword: '', content: '' }]);
+  const [summary, setSummary] = useState('');
+
+  const handleAddSection = () => {
+    const newSection = { id: shortid.generate(), keyword: '', content: '' };
+    setSections(sections.concat(newSection));
+  };
+
+  const handleRemoveSection = (e) => {
+    const sectionId = e.currentTarget.dataset.target;
+
+    if (sections.length === 1) return;
+
+    if (sectionId) {
+      const index = sections.findIndex((section) => section.id === sectionId);
+      setSections([...sections.slice(0, index), ...sections.slice(index + 1, sections.length)]);
```

Comment:
Using filter would make this a lot more readable 😄 
```suggestion
      setSections(sections.filter(section => section.id !== sectionId));
```